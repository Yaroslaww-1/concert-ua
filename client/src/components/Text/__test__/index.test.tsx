import React from 'react';
import { render } from '@testing-library/react';
import TextComponent from '../index';

describe('TextComponent', () => {
  it('should match snapshot', () => {
    const text = 'Test text';
    const { container } = render(<TextComponent>{text}</TextComponent>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render div with the text', () => {
    const text = 'Test text';
    const { container } = render(<TextComponent>{text}</TextComponent>);
    const element = container.getElementsByTagName('div').item(0);
    expect(element).not.toBeNull();
    expect(element!.innerHTML).toBe(text);
  });
});

// test('Link changes the class when hovered', () => {
//   // let tree = component.toJSON();
//   // expect(tree).toMatchSnapshot();
//   // // manually trigger the callback
//   // tree.props.onMouseEnter();
//   // // re-rendering
//   // tree = component.toJSON();
//   // expect(tree).toMatchSnapshot();
//   // // manually trigger the callback
//   // tree.props.onMouseLeave();
//   // // re-rendering
//   // tree = component.toJSON();
//   // expect(tree).toMatchSnapshot();
// });
