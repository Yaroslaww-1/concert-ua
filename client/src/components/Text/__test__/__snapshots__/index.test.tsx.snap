// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextComponent should match snapshot 1`] = `
<div
  class="root "
  style="font-size: 1rem; text-transform: none; line-height: 1.4; word-break: break-all; text-align: center; font-weight: 400; letter-spacing: normal; font-family: Roboto;"
>
  Test text
</div>
`;

exports[`TextComponent should render something 1`] = `
<div
  class="root "
  style="font-size: 1rem; text-transform: none; line-height: 1.4; word-break: break-all; text-align: center; font-weight: 400; letter-spacing: normal; font-family: Roboto;"
>
  Test text
</div>
`;

exports[`TextComponent should render text 1`] = `
<div
  class="root "
  style="font-size: 1rem; text-transform: none; line-height: 1.4; word-break: break-all; text-align: center; font-weight: 400; letter-spacing: normal; font-family: Roboto;"
>
  Test text
</div>
`;
